### ISSUE TITLE
`if_not_else` false positive with macro condition

### ISSUE THREAD

**Issue by DanielEScherzer** [rust-lang/rust-clippy#17373]
### Summary

The if_not_else lint appears to misunderstand conditions from a macro, with the suggestion to repeat the condition (replace `if HT_IS_PACKED!(self)` with `if HT_IS_PACKED!(self) == HT_IS_PACKED!(self)` which is then caught by the `eq_op` lint) and flip the conditional code blocks (which is normally the correct solution to `if_not_else`, but the lint should not be triggered here)

### Lint Name

if_not_else

### Reproducer

I tried this code:

```rust
#![allow(dead_code)]
#![deny(clippy::if_not_else)]

macro_rules! HT_IS_PACKED {
    ($ht:expr) => {
        ($ht).u & 4 != 0
    };
}

struct HashTable {
    u: u32,
}

impl HashTable {
    fn check(&mut self) {
        if HT_IS_PACKED!(self) {
            println!("Packed");
        } else {
            println!("Not packed");
        }
    }
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
error: unnecessary `!=` operation
  --> src/lib.rs:16:9
   |
16 | /         if HT_IS_PACKED!(self) {
17 | |             println!("Packed");
18 | |         } else {
19 | |             println!("Not packed");
20 | |         }
   | |_________^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.96.0/index.html#if_not_else
note: the lint level is defined here
  --> src/lib.rs:2:9
   |
 2 | #![deny(clippy::if_not_else)]
   |         ^^^^^^^^^^^^^^^^^^^
help: try
   |
16 ~         if HT_IS_PACKED!(self) == HT_IS_PACKED!(self) {
17 +             println!("Not packed");
18 +         } else {
19 +             println!("Packed");
20 +         }
   |

error: could not compile `playground` (lib) due to 1 previous error
```

I expected to see this happen:

no error is output

### Version

```text
Playground 1.96.1
```

### Additional Labels

@rustbot label +I-suggestion-causes-bug
